python/pfs/drp/stella/fitPfsFluxReference.py

```diff
@@ -1996,6 +2021,42 @@ def removeBadFluxes(
     pfsConfig.totalFluxErr = totalFluxErrArrays
 
 
+def removeBadSpectra(
+    pfsMerged: PfsMerged,
+    cutoffSNR: float,
+    cutoffSNRRange: tuple[float, float],
+) -> PfsMerged:
+    """Remove bad spectra from ``pfsMerged``.
+
+    Spectra are bad if their average pixel-wise S/N is worse than ``cutoffSNR``.
+
+    Parameters
+    ----------
+    pfsMerged : `PfsMerged`
+        Merged spectra from exposure.
+    cutoffSNR : `float`
+        Cut-off S/N.
+    cutoffSNRRange : `tuple` [`float`, `float`]
+        Wavelength range (nm) in which pixel-wise S/N is averaged to be compared
+        with ``cutoffSNR``.
+
+    Returns
+    -------
+    pfsMerged : `PfsMerged`
+        PfsMerged from which bad spectra have been removed.
+    """
+    left, right = cutoffSNRRange
+    good = np.zeros(shape=(len(pfsMerged),), dtype=bool)
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        for i in range(len(pfsMerged)):
+            sampleIndex = (left <= pfsMerged.wavelength[i]) & (pfsMerged.wavelength[i] <= right)
+            snr = pfsMerged.flux[i, sampleIndex] / np.sqrt(pfsMerged.covar[i, 0, sampleIndex])
```

Comment:
Maybe `pfsMerged.variance[i, sampleIndex]`?

Reply:
I am adding another commit to this PR, in which I replace all occurrences of "covar" with "variance" in the entire flux calibration code.